api/rest.go

```diff
@@ -1980,6 +1984,26 @@ func handleContentUnitsList(cp utils.ContextProvider, exec boil.Executor, r Cont
 	}, nil
 }
 
+func findSourceForSourceTypes(exec boil.Executor, units []*models.ContentUnit) ([]*models.Source, error) {
+	var ids []int64
+	for _, u := range units {
+		if u.TypeID != common.CONTENT_TYPE_REGISTRY.ByName[common.CT_SOURCE].ID {
+			ids = append(ids, u.ID)
+		}
+	}
+
+	sources, err := models.Sources(exec,
+		qm.InnerJoin("content_unit_source cus ON cus.source_id = s.id"),
+		qm.WhereIn("cus.content_unit_id IN ?", utils.ConvertArgsInt64(ids)...),
+		qm.Load("SourcesI18ns", "ContentUnits")).
```

Comment:
loading all units associated with a source has the potential to load to much data. 
Is there no other way ? Is it really must be like this ?